builtin/logical/rabbitmq/path_role_create.go

```diff
@@ -0,0 +1,111 @@
+package rabbitmq
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/uuid"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"github.com/michaelklishin/rabbit-hole"
+)
+
+func pathRoleCreate(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "creds/" + framework.GenericNameRegex("name"),
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Name of the role.",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation: b.pathRoleCreateRead,
+		},
+
+		HelpSynopsis:    pathRoleCreateReadHelpSyn,
+		HelpDescription: pathRoleCreateReadHelpDesc,
+	}
+}
+
+func (b *backend) pathRoleCreateRead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	name := data.Get("name").(string)
+
+	// Get the role
+	role, err := b.Role(req.Storage, name)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("unknown role: %s", name)), nil
+	}
+
+	// Determine if we have a lease
+	lease, err := b.Lease(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if lease == nil {
+		lease = &configLease{}
+	}
+
+	displayName := req.DisplayName
+	if len(displayName) > 26 {
+		displayName = displayName[:26]
+	}
+	userUUID := uuid.GenerateUUID()
+	username := fmt.Sprintf("%s-%s", displayName, userUUID)
+	if len(username) > 63 {
+		username = username[:63]
+	}
+	password := uuid.GenerateUUID()
+
+	// Get our connection
+	client, err := b.Client(req.Storage)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I realize that `b.Client` won't return a nil client with your current code, but for future safety's sake, it'd be good to ensure that `client != nil` here.